extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java

```diff
@@ -1061,6 +1068,96 @@ public List<HandlerChainCustomizer> scan(MethodInfo method, ClassInfo actualEndp
+    @BuildStep
+    io.quarkus.arc.deployment.AnnotationsTransformerBuildItem removeStandardSecurityAnnotationsFromResources(
+            Capabilities capabilities,
+            Optional<ResourceScanningResultBuildItem> resourceScanningResultBuildItem,
+            ApplicationResultBuildItem applicationResultBuildItem, BeanArchiveIndexBuildItem beanArchiveIndexBuildItem) {
+
+        if (!capabilities.isPresent(Capability.SECURITY) || resourceScanningResultBuildItem.isEmpty()) {
+            return null;
+        }
+
+        final IndexView index = beanArchiveIndexBuildItem.getIndex();
+
+        // collect all endpoints
+        final Set<MethodInfo> endpoints = findEndpoints(resourceScanningResultBuildItem.get().getResult(), index,
+                applicationResultBuildItem.getResult());
+
+        final Set<DotName> visitedClasses = new HashSet<>();
+        final Set<DotName> annotatedClasses = new HashSet<>();
+        final Set<MethodDescriptor> annotatedMethods = new HashSet<>();
+        for (MethodInfo endpoint : endpoints) {
+
+            // collect endpoints annotated with a security annotation
+            if (SecurityTransformerUtils.hasStandardSecurityAnnotation(endpoint)) {
+                annotatedMethods.add(MethodDescriptor.of(endpoint));
+            }
+
+            // collect resource (super) classes annotated with a security annotation
+            ClassInfo c = endpoint.declaringClass();
+            while (c.superName() != null) {
+                if (!visitedClasses.contains(c.name())) {
+                    visitedClasses.add(c.name());
+                    if (SecurityTransformerUtils.hasStandardSecurityAnnotation(c)) {
+                        annotatedClasses.add(c.name());
+                    }
+                }
+                c = index.getClassByName(c.superName());
+            }
+        }
+
+        final boolean appliesToClasses = !annotatedClasses.isEmpty();
+        final boolean appliesToMethods = !annotatedMethods.isEmpty();
+        final Predicate<AnnotationTarget.Kind> appliesTo;
+        if (appliesToClasses && appliesToMethods) {
+            appliesTo = Predicate.<AnnotationTarget.Kind> isEqual(AnnotationTarget.Kind.METHOD)
+                    .or(Predicate.isEqual(AnnotationTarget.Kind.CLASS));
+        } else if (appliesToMethods) {
+            appliesTo = Predicate.isEqual(AnnotationTarget.Kind.METHOD);
+        } else if (appliesToClasses) {
+            appliesTo = Predicate.isEqual(AnnotationTarget.Kind.CLASS);
+        } else {
+            // found no security annotations
+            return null;
+        }
+
+        final Predicate<AnnotationInstance> isSecurityAnnotation = new Predicate<AnnotationInstance>() {
+            @Override
+            public boolean test(AnnotationInstance annotationInstance) {
+                return SecurityTransformerUtils.SECURITY_BINDINGS.contains(annotationInstance.name());
+            }
+        };
+        // remove security annotations
```

Comment:
@michalvavrik Hi, one thing I'd like to clarify, can it affect the custom interceptors (not those provided by `quarkus-security` but the ones added by the users) ?

Reply:
@sberyozkin absolutely! However Martin and Georgios thinks it's unlikely anyone is binding custom interceptors to security  annotations.

Reply:
it at least that's how I understood our communication above, please see previous comments.

Reply:
I'll need to review this thorougly

Reply:
Hmm, it sounds like it can be a breaking change. @knutwannheden in #26536's description refers to a custom interceptor doing the security logging, but I don't know how it is bound. But it feels a bit risky.

So re your earlier comment: 
```
In short -
annotation transformer solution: one annotation instance - possibly many interceptors -> you remove it -> you loose all the bindings
this pr: one annotation instance - possibly many interceptor -> you apply filter -> you loose the binding between an interceptor and the method (so all other bindings are still intercepted). I only implemented what I really need so non-static invocation interceptions of methods)
```
I think your original solution should've stayed, it was hard for me to suggest something useful as the Arc code was quite complex, but if that complexity was needed to support the custom interceptors then it should be preferred IMHO.

However lets wait for more comments.